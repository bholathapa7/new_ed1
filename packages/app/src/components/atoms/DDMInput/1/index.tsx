import styled from 'styled-components';

import palette from '^/constants/palette';

export interface Props {
  readonly error?: boolean;
}
const DDMInput = styled.input<Props>(({ error }) => ({
  boxSizing: 'border-box',

  width: '100%',
  height: '40px',
  padding: '0 20px',

  borderWidth: '1px',
  borderStyle: 'solid',
  borderRadius: '3px',

  fontSize: '13px',
  fontWeight: 'normal',
  color: palette.textGray.toString(),

  '::placeholder': {
    color: palette.textLight.toString(),
  },

  borderColor: (error ? palette.error : palette.border).toString(),
}));

export default DDMInput;
