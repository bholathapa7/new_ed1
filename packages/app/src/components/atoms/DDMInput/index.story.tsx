import { storiesOf } from '@storybook/react';
import React from 'react';
import styled from 'styled-components';

import palette from '^/constants/palette';
import { DDMInput } from '.';

const Wrapper = styled.div({
  width: '287px',
  height: '37px',

  padding: '50px',
  backgroundColor: palette.white.toString(),
});

storiesOf('Atoms|V2_DDMInput', module)
  .add('default', () => (
    <Wrapper>
      <DDMInput hasError={false} placeholder='PLACEHOLDER' />
    </Wrapper>
  )).add('error', () => (
    <Wrapper>
      <DDMInput hasError={true} placeholder='PLACEHOLDER' />
    </Wrapper>
  ));
