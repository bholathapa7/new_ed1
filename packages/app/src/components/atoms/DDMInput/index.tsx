import styled from 'styled-components';

import dsPalette from '^/constants/ds-palette';
import palette from '^/constants/palette';

export interface Props {
  hasError?: boolean;
}

export const DDMInput = styled.input<Props>(({ hasError }) => ({
  boxSizing: 'border-box',

  width: '100%',
  height: '100%',
  paddingLeft: '12px',

  border: `1px solid ${(hasError ? palette.DDMInput.error : palette.DDMInput.inputBorder).toString()}`,
  borderRadius: '5px',
  color: hasError ? palette.DDMInput.error.toString() : dsPalette.title.toString(),
  fontSize: '13px',

  // eslint-disable-next-line no-magic-numbers
  backgroundColor: hasError ? palette.DDMInput.error.alpha(0.05).toString() : undefined,

  '::placeholder': {
    color: (hasError ? palette.DDMInput.error : palette.dividerLight).toString(),
  },
}));
