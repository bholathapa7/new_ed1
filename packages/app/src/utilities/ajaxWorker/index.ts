import { Observable, from } from 'rxjs';
import * as RxjsAjax from 'rxjs/ajax';
import { AjaxCreationMethod } from 'rxjs/internal/observable/dom/AjaxObservable';
import Worker from 'worker-loader!./worker';
import { AJAX_METHOD, AjaxMethod, AjaxParams } from './types';

function createAjaxFromWorker(w: Worker, method: AjaxMethod): Observable<AjaxResponse> {
  return from(new Promise<AjaxResponse>((resolve, reject) => {
    w.onmessage = ({ data }: MessageEvent) => {
      const resp: AjaxResponse = method === AJAX_METHOD.RAW ? data : JSON.parse(data);

      resolve(resp);
      w.terminate();
    };
    w.onerror = (msgEvent: ErrorEvent) => {
      reject(JSON.parse(msgEvent.error));
      w.terminate();
    };
  }));
}


function initializeWorker(ajaxParams: AjaxParams | (string | AjaxRequest), method: AjaxMethod): Observable<AjaxResponse> {
  const worker: Worker = new Worker();
  worker.postMessage(JSON.stringify({ ajaxParams, method }));

  return createAjaxFromWorker(worker, method);
}

export const ajax: {
  get: AjaxCreationMethod['get'];
  post: AjaxCreationMethod['post'];
  delete: AjaxCreationMethod['delete'];
  patch: AjaxCreationMethod['patch'];
  basic(urlOrRequest: string | AjaxRequest): Observable<AjaxResponse>;
  raw(urlOrRequest: string | AjaxRequest): Observable<AjaxResponse>;
} = {
  raw: (urlOrRequest) => initializeWorker(urlOrRequest, AJAX_METHOD.RAW),
  basic: (urlOrRequest) => initializeWorker(urlOrRequest, AJAX_METHOD.BASIC),
  get: (url, headers) => initializeWorker({ url, headers }, AJAX_METHOD.GET),
  post: (url, body, headers) => initializeWorker({ url, headers, body }, AJAX_METHOD.POST),
  delete: (url, headers) => initializeWorker({ url, headers }, AJAX_METHOD.DELETE),
  patch: (url, body, headers) => initializeWorker({ url, headers, body }, AJAX_METHOD.PATCH),
};

export type AjaxError = RxjsAjax.AjaxError;
export type AjaxRequest = RxjsAjax.AjaxRequest;
export type AjaxResponse = RxjsAjax.AjaxResponse;
