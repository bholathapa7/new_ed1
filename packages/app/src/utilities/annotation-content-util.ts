import * as _ from 'lodash-es';
import { Coordinate } from 'ol/coordinate';

import { CONTENT_NUMBER_GAP, CONTENT_TITLE_PREFIX, START_CONTENT_NUMBER, START_MEASUREMENT_NUMBER } from '^/constants/defaultContent';
import { L10nFn } from '^/hooks';
import * as T from '^/types';
import { contentTexts } from './content-util';

export const getOrderedTitle: (name: T.Content['title'], usingNames: Array<T.Content['title']>) => T.Content['title'] = (
  name, usingNames,
) => {
  const postfix: Array<number> = usingNames
    // Contents must start with the exact name
    .filter((n) => n.includes(name) && n.indexOf(name) === 0)
    // Only retrieve anything else after the name to increment the number.
    .map((n) => n.replace(name, '').match(/\s*\d*$/))
    .map((m) => m === null ? START_MEASUREMENT_NUMBER : Number(m[0]));
  const usingOrdered: Array<number> = _.uniq(postfix);
  const maxOrder: number | undefined = _.max(usingOrdered);

  /*
   * I tried to start newOrder from 1 and not display 1 in a way.
   * However, if the user changes to 1 directly, it starts from 0 because it needs to branch.
   */
  let newOrder: number = START_MEASUREMENT_NUMBER;
  while (maxOrder !== undefined && newOrder <= maxOrder && usingOrdered.includes(newOrder)) {
    if (newOrder === START_MEASUREMENT_NUMBER) {
      newOrder++;
    }

    newOrder++;
  }

  return usingOrdered.length && newOrder > START_MEASUREMENT_NUMBER ? `${name} ${newOrder}` : name;
};

export const getCopiedContentTitle: (
  selectedContentTitle: T.Content['title'],
  usingContentTitles: Array<T.Content['title']>
) => T.Content['title'] = (
  selectedContentTitle,
  usingContentTitles,
) => {
  const postfixOfContentsWithSameType: Array<number> = usingContentTitles
    .filter((title) => title.includes(selectedContentTitle) && title.indexOf(selectedContentTitle) === 0)
    .map((title) => {
      const postfix = title.replace(selectedContentTitle, '');
      const parsedNum = Number(postfix.match(/\(([^)]+)\)/)?.pop()) ?? START_CONTENT_NUMBER;
      return parsedNum;
    })
    .filter((num) => !isNaN(num))
    .sort((a, b) => a - b);
  const nextContentNum: number = postfixOfContentsWithSameType.length
    ? postfixOfContentsWithSameType[postfixOfContentsWithSameType.length - 1] + CONTENT_NUMBER_GAP
    : START_CONTENT_NUMBER + CONTENT_NUMBER_GAP;

  return `${selectedContentTitle}${CONTENT_TITLE_PREFIX}(${nextContentNum})`;
};

export const getNotDuplicatePoints: (locations: Array<Coordinate>) => Array<Coordinate> = (
  locations,
) => {
  const refinedLocations: Array<Coordinate> = [];
  locations
    .map((p) => p.toString())
    .forEach((hash, i) => {
      const lastPoint: Coordinate | undefined = _.last(refinedLocations);
      if (lastPoint && lastPoint.toString() === hash) {
        return;
      }
      refinedLocations.push(locations[i]);
    });

  return refinedLocations;
};

export const getContentTitle: (content: T.Content, l10n: L10nFn) => T.Content['title'] = (content, l10n) => {
  switch (content.type) {
    case T.ContentType.MAP:
    case T.ContentType.DSM:
    case T.ContentType.POINTCLOUD:
    case T.ContentType.THREE_D_ORTHO:
    case T.ContentType.GCP_GROUP:
    case T.ContentType.THREE_D_MESH:
      return l10n(contentTexts[content.type]);

    default:
      return content.title;
  }
};

export const getHasColor: (contentType: T.ContentType) => boolean = (
  contentType,
) => [
  T.ContentType.MARKER,
  T.ContentType.LENGTH,
  T.ContentType.AREA,
  T.ContentType.VOLUME,
  T.ContentType.ESS_ARROW,
  T.ContentType.ESS_POLYLINE,
  T.ContentType.ESS_POLYGON,
].includes(contentType);
