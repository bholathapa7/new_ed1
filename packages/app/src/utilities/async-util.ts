import * as Sentry from '@sentry/browser';

export interface TryCatchOutput<T> {
  data: T | null;
  error: Error | null;
}

/**
 * Abstraction to make async actions easier.
 */
export async function tryCatch<T>(promise: Promise<T>): Promise<TryCatchOutput<T>> {
  try {
    const data: T = await promise;

    return { data, error: null };
  } catch (error) {
    Sentry.captureException(error);

    return { error, data: null };
  }
}
