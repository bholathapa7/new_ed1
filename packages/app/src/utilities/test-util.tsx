import { createLocation } from 'history';
import createMockStore, { MockStoreCreator, MockStoreEnhanced } from 'redux-mock-store';
import { action as makeAction, payload, union } from 'tsdux';

import * as T from '^/types';

import * as M from '^/store/Mock';

const mockStoreCreator: MockStoreCreator<any> = createMockStore<any>();

/**
 * @author Joon-Mo Yang <jmyang@angelswing.io>
 * @desc mock context creator for common page state
 */

export function mockCommonContext() {
  const commonStore = {
    Pages: {
      Common: M.mockCommonPage,
    },
  };

  return {
    store: (mockStoreCreator as MockStoreCreator<typeof commonStore>)(commonStore),
  };
}


export const UpdateLanguage = makeAction(
  'test/UPDATE_LANGUAGE',
  payload<T.Language>(),
);

export const Auth = makeAction(
  'test/AUTH',
);
export const Unauth = makeAction(
  'test/UNAUTH',
);

const Action = union([
  UpdateLanguage,
  Auth,
  Unauth,
]);
export type DDMMockAction = typeof Action;


export type DDMMockStore = MockStoreEnhanced<T.State>;

export const initialMockState: T.State = {
  Attachments: M.mockAttachments,
  Auth: M.mockAuth,
  Contents: M.mockContents,
  Screens: M.mockScreens,
  Pages: M.mockPages,
  Permissions: M.mockPermissions,
  Projects: M.mockProjects,
  SharedContents: M.mockSharedContents,
  Users: M.mockUsers,
  UserConfig: M.mockUserConfig,
  PlanConfig: M.mockPlanConfig,
  ESSModels: M.mockESSModels,
  ESSContents: M.mockESSContents,
  ESSAttachments: M.mockESSAttachments,
  ProjectConfigPerUser: M.mockProjectConfigPerUser,
  Photos: M.mockPhotos,
  Groups: M.mockGroups,
  router: {
    location: createLocation(''),
    action: 'PUSH',
  },
};

export const createDDMMockStore: (
  initialLanguage: T.Language,
) => DDMMockStore = (
  initialLanguage,
) => {
  const initialState: T.State = {
    ...initialMockState,
    Pages: {
      ...M.mockPages,
      Common: {
        ...M.mockCommonPage,
        language: initialLanguage,
      },
    },
    Permissions: M.mockPermissions,
    Projects: M.mockProjects,
    SharedContents: M.mockSharedContents,
    Users: M.mockUsers,
    router: {
      location: createLocation(''),
      action: 'PUSH',
    },
    ProjectConfigPerUser: {
      config: undefined,
      patchProjectConfigStatus: T.APIStatus.IDLE,
    },
    PlanConfig: M.mockPlanConfig,
  };

  return mockStoreCreator((
    actions: Array<DDMMockAction>,
  ) => actions.reduce((state, action) => {
    switch (action.type) {
      case UpdateLanguage.type:
        return {
          ...state,
          Pages: {
            ...state.Pages,
            Common: {
              ...state.Pages.Common,
              language: action.payload,
            },
          },
        };
      case Auth.type:
        return {
          ...state,
          Auth: {
            ...state.Auth,
            authedUser: {
              id: state.Users.users.allIds[0],
              token: 'random-test-token',
            },
          },
        };
      case Unauth.type:
        return {
          ...state,
          Auth: {
            ...state.Auth,
            authedUser: undefined,
          },
        };
      default:
        return state;
    }
  }, initialState));
};

/**
 * @desc common `afterEach` hook for mount testing
 */
export function commonAfterEach(): void {
  // @ts-ignore
  jest.clearAllTimers();
  // @ts-ignore
  jest.resetAllMocks();
}
