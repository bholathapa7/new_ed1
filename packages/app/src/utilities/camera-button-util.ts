import download from '^/utilities/download';
import html2canvas from 'html2canvas';

import * as T from '^/types';
import { TryCatchOutput, tryCatch } from '^/utilities/async-util';
import { getUserAgent } from './userAgent';

export const runScreenCapture: (element: HTMLElement, capturedFileName: string) => Promise<void> =
  async (element, capturedFileName) => {
    /**
     * @description When browser is webkit based
     * and usePopup parameter of download function is false, then the browser redirects to blob url.
     */
    const isWebkit: boolean = getUserAgent() === T.UserAgent.SAFARI;

    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
    const { data, error }: TryCatchOutput<HTMLCanvasElement> = await tryCatch(html2canvas(element, {
      allowTaint: true,
      useCORS: true,
    }) as Promise<HTMLCanvasElement>);

    if (!error && data) {
      download(data.toDataURL(), isWebkit, true, capturedFileName);
    }
  };

interface TwoDHandleCameraClickParams {
  onStart(): void;
  onEnd(): void;
  mainScreenCaptureFunction(): Promise<void>;
}

type TwoDHandleCameraClick = ({
  onStart,
  onEnd,
  mainScreenCaptureFunction,
}: TwoDHandleCameraClickParams) => () => Promise<void>;

export const handleCameraClickFunctor: TwoDHandleCameraClick = ({
  onStart,
  onEnd,
  mainScreenCaptureFunction,
}) => async () => {
  onStart();
  await mainScreenCaptureFunction();
  onEnd();
};
