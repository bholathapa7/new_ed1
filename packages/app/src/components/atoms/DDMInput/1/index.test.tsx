import * as ReactTesting from '@testing-library/react';
import React from 'react';

import { commonAfterEach } from '^/utilities/test-util';

import DDMInput from '.';

describe('DDMInput', () => {
  let renderResult: ReactTesting.RenderResult;

  beforeEach(() => {
    renderResult = ReactTesting.render(
      <DDMInput error={false} />,
    );
  });

  afterEach(ReactTesting.cleanup);
  afterEach(commonAfterEach);

  it('should have valid style for errorless state', () => {
    renderResult.rerender(
      <DDMInput error={false} />,
    );
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it('should have valid style for error state', () => {
    renderResult.rerender(
      <DDMInput error={true} />,
    );
    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
