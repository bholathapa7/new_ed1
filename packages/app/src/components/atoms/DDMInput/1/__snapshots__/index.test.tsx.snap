// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DDMInput should have valid style for error state 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 100%;
  height: 40px;
  padding: 0 20px;
  border-width: 1px;
  border-style: solid;
  border-radius: 3px;
  font-size: 13px;
  font-weight: normal;
  color: rgb(102,102,102);
  border-color: rgb(252,74,74);
}

.c0::-webkit-input-placeholder {
  color: rgb(153,153,153);
}

.c0::-moz-placeholder {
  color: rgb(153,153,153);
}

.c0:-ms-input-placeholder {
  color: rgb(153,153,153);
}

.c0::placeholder {
  color: rgb(153,153,153);
}

<input
    class="c0"
  />
</DocumentFragment>
`;

exports[`DDMInput should have valid style for errorless state 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 100%;
  height: 40px;
  padding: 0 20px;
  border-width: 1px;
  border-style: solid;
  border-radius: 3px;
  font-size: 13px;
  font-weight: normal;
  color: rgb(102,102,102);
  border-color: rgb(219,219,219);
}

.c0::-webkit-input-placeholder {
  color: rgb(153,153,153);
}

.c0::-moz-placeholder {
  color: rgb(153,153,153);
}

.c0:-ms-input-placeholder {
  color: rgb(153,153,153);
}

.c0::placeholder {
  color: rgb(153,153,153);
}

<input
    class="c0"
  />
</DocumentFragment>
`;
